src/vm/vm.cpp

```diff
@@ -488,12 +507,18 @@ void RunThread(Sc3VmThread* thread) {
       if (opcodeGrp1 == 0x10) {
         OpcodeTableUser1[opcode](thread);
       } else if (opcodeGrp1 == 0x02) {
-        OpcodeTableGraph3D[opcode](thread);
```

Comment:
You've removed 3D instructions thus breaking all R;Ns.
Let's not do it like this, let's do something like
```
else if (opcodeGrp1 == 0x02 && OpcodeTableGraph3D != nullptr) {
  OpcodeTableGraph3D[opcode](thread);
}
.....
```
Games that don't have 3D will never hit this if every instruction is defined properly on our side. I don't think a special message is needed for this as the reason is the same as it is in "Unknown opcode" case. This would also get rid of "goto" which I don't think is necessary here.

Reply:
Whoops, must've been left over from when I was debugging a related issue. Fixed.